### PR TITLE
SG Minigun Definitive Edition

### PR DESCRIPTION
**Pull request by Arctrooper8**
## About The Pull Request

Finally, the perfect version. Denied my test merge time I have been forced to experiment and probe with PRs, constantly tweaking and adjusting. But now, perfection awaits!

#### SG MINIGUN:
Fire rate .15 -> .1
Windup delay .4 -> .7
Damage 15 -> 10

#### AMMO:
SG-85 powerpack ammo 1000 -> 3000
SG ammo box added, holds 1000 rounds
SG-29 drum ammo 200 -> 250
Fixes : #10317

#### NIGHT VISION:
All night vision goggles now use LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE (e.g. Imagers with mesonness)

#### FACTORY:
(This applies for both SG-29 and SG-85 ammo production)
A refill produces 10 final products
A refill costs 25 points (purchasing all mags/boxes individually would cost 50 points)
Fixes : #10226

## Why It's Good For The Game

The SG minigun was merged with no TM time at all. What followed was a couple of reactionary kneejerk PRs that changed the minigun for better (sound) or worse (removing it's unique identity).

#### FIRERATE:
The biggest change will be the return to a .1 fire rate which is better for a plethora of reasons:
First: .1 fire rate gives the SG minigun a unique feel and role making it the ultimate suppressive weapon.
Two: We have already had weapons with .1 fire rate and not a single claim of them being laggy or causing issues. (In fact the shotguns seem to cause the most projectile based lag spikes and complaints)
Three: The SG minigun has already had .1 fire rate before for a decent time, people I have talked to with bad to great computers/connections have made little complaint about the SG minigun causing lag. In fact, many people were upset when it changed from .1 to .15 already having fond memories of the high fire rate in such a short time frame.
Overall changing the fire rate back to .1 won't be a major issue and will help the SG minigun return back to its intendent niche and generally just being buckets of fun.

#### AMMO:
Since the SG minigun will have a much higher fire rate it is gonna need a lot more bullets.
The ammo boxes serve as a unique way for the SG-85 to be reloaded which forces SGs to disconnect their minigun and hold the pack in hand to refill it with the box. This allows SGs to take more ammo, and Xenos opportunities to counter play.
When the SG-29 got its higher fire rate it struggled to keep up Ammo wise, this should allow for smoother SG-29 gameplay without being a major balance change.

#### NIGHT VISION:
Right now the SG headset fully brightens in a 4 tile radius around you (the variable might be five but that first tile seems to be the one you stand on and thus barely does anything) it also gives see turfs vision similar to mesons. Changing this to LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE results in them acting like more powerful optical imagers, this is good for a couple of reasons:
1. SGs are more aware of light such as flares, ally suit lights, or their own suit light
2. LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE arguably looks better compared to the current SG goggles where there is fully brightened tiles and then pitch black squares where your night vision does not reach.
3. SG having free night vision goggles has always been a tad odd, however as the squad suppressive fire specialists it mostly made sense but Xenos should by far be the superior predators in the dark.

In conclusion I hoped you enjoyed reading this Essay as much as I enjoyed writing it. 
On a serious note the SG is a suppressive fire support role, making the SG-85 simply an IFF minigun was a mistake which we have seen before where the T-25 was power crept into something absurd, we should strive to make the SG-85 a unique weapon that is fun to use but not abhorrent to play against or unbalanced.

## Changelog
:cl:
add: SG minigun now has ammo boxes which hold 1000 rounds each.
balance: SG minigun is now perfect and nobody will ever have to balance it again.
balance: SG ammo factory refills are now cheaper but produce less.
balance: SG-29 drum now holds 250.
balance: All night vision goggles now use LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
fix: You can no longer buy T-25 refills
/:cl:

### CODE REVIEW

**Review by Pariah919 — COMMENTED** [review]
Looks good to me but I dislike the NVG changes, removes the mesonness out of them.

**Comment by Wayland-Smithy** [comment]
With how lighting works the vision buff is not to be *overlooked* here. Basically you will have +2 tiles beyond the end of the visible light and full vision in some places because of lighting jank.

**Comment by Al-1ce** [comment]
> Looks good to me but I dislike the NVG changes, removes the mesonness out of them.

Agreed.

**Review by BraveMole — CHANGES_REQUESTED** [review]
Do what mjp said and i merge

**Comment by Arctrooper8** [comment]
Apologies for the late response, I haven't been able to access my main computer recently.
First off, thank you for the excellent feedback, I have compiled some facts/images to further clarify the NVG changes.

1. SG goggles still have "Mesonness" due to the fact that being controlled by vision_flags = SEE_TURFS which I have not removed.
2. I have removed the range buff as I was unaware of it's true strength.

Crash site three

Now for some helpful images of what the change will look like (as you can see the SG goggles still have their Mesonness)
#### No goggles
No light:
image

Suit light:
image

#### Current SG goggles (I had to do these during a new session because I messed up the code)
No light:
image

Suit light:
image

#### Proposed SG goggles change
No light:
image

Suit light:
image

**Comment by Arctrooper8** [comment]
> Do what mjp said and i merge

Done, the big picture comment gives more clarification if needed.

**Review by BraveMole — APPROVED** [review]

**Comment by Losenis** [comment]
Can't load the powerpack with ammo bins, but you can load ammo bins with the power pack

**Comment by Arctrooper8** [comment]
> Can't load the powerpack with ammo bins, but you can load ammo bins with the power pack

Thanks for pointing this out, I forgot to give the powerpack the refillable flag when I moved this over to a new branch a while ago. I have tested it on local and it works now.
